packages/data/src/redux-store/metadata/reducer.ts

```diff
@@ -13,15 +13,19 @@ import { selectorArgsToStateKey, onSubKey } from './utils';
 type Action =
 	| ReturnType< typeof import('./actions').startResolution >
 	| ReturnType< typeof import('./actions').finishResolution >
+	| ReturnType< typeof import('./actions').failResolution >
 	| ReturnType< typeof import('./actions').startResolutions >
 	| ReturnType< typeof import('./actions').finishResolutions >
+	| ReturnType< typeof import('./actions').failResolutions >
 	| ReturnType< typeof import('./actions').invalidateResolution >
 	| ReturnType< typeof import('./actions').invalidateResolutionForStore >
 	| ReturnType<
 			typeof import('./actions').invalidateResolutionForStoreSelector
 	  >;
 
-export type State = EquivalentKeyMap< unknown[] | unknown, boolean >;
+type StateKey = unknown[] | unknown;
+export type StateValue = { isResolving: boolean; error?: Error | unknown };
```

Comment:
I wonder if it makes sense to just use the `status` enum in the reducer rather than using `isResolving` boolean? So that we can keep the state as simple as possible, higher-level selectors can interpret the value and expose `getIsResolving` or `getHasResolved` from there.

Reply:
@kevin940726 Let me also loop in @youknowriad @jsnajdr and @gziolo, it would be great to get this one in

Reply:
@youknowriad I was pretty cryptic, wasn't I? :D Would you mind reviewing the PR? It would be great to get it shipped.